x-pack/plugins/files/public/components/image/image.tsx

```diff
@@ -28,22 +45,59 @@ export interface Props extends ImgHTMLAttributes<HTMLImageElement> {
  * ```
  */
 export const Image = React.forwardRef<HTMLImageElement, Props>(
-  ({ src, alt, onFirstVisible, ...rest }, ref) => {
+  (
+    { src, alt, onFirstVisible, onLoad, onError, meta, wrapperProps, size = 'original', ...rest },
+    ref
+  ) => {
+    const [isLoaded, setIsLoaded] = useState<boolean>(false);
+    const [blurDelayExpired, setBlurDelayExpired] = useState(false);
     const { isVisible, ref: observerRef } = useViewportObserver({ onFirstVisible });
+
+    useEffect(() => {
+      const id = window.setTimeout(() => setBlurDelayExpired(true), 200);
+      return () => window.clearTimeout(id);
+    }, []);
+
+    const knownSize = size ? sizes[size] : undefined;
+
     return (
-      <img
-        {...rest}
-        ref={(element) => {
-          observerRef(element);
-          if (ref) {
-            if (typeof ref === 'function') ref(element);
-            else (ref as MutableRefObject<HTMLImageElement | null>).current = element;
-          }
-        }}
-        // TODO: We should have a lower resolution alternative to display
-        src={isVisible ? src : undefined}
-        alt={alt}
-      />
+      <div
+        css={[
+          css`
+            position: relative;
+            display: inline-block;
+          `,
+          knownSize,
+        ]}
+        {...wrapperProps}
+      >
+        {blurDelayExpired && meta?.blurhash && (
+          <Blurhash
+            visible={!isLoaded}
+            hash={meta.blurhash}
+            isContainerWidth={size !== 'original' && size !== undefined}
+            width={meta.width}
+            height={meta.height}
+          />
+        )}
+        <MyImage
```

Comment:
nit: my/your maybe `<BaseImage`

Reply:
Haha, you never had a MyJavaApplication?